Std/Data/Bool.lean

```diff
@@ -0,0 +1,251 @@
+  decide
+
+theorem and_xor_distrib_left : ∀ (x y z : Bool), (x && xor y z) = xor (x && y) (x && z) := by
+  decide
+
+theorem and_xor_distrib_right : ∀ (x y z : Bool), (xor x y && z) = xor (x && z) (y && z) := by
+  decide
+
+theorem and_deMorgan : ∀ (x y : Bool), (!(x && y)) = (!x || !y) := by
+  decide
+
+theorem and_eq_true_iff : ∀ (x y : Bool), (x && y) = true ↔ x = true ∧ y = true := by
+  decide
+
+theorem and_eq_false_iff : ∀ (x y : Bool), (x && y) = false ↔ x = false ∨ y = false := by
+  decide
+
+alias and_false_left := Bool.false_and
+alias and_false_right := Bool.and_false
+alias and_true_left := Bool.true_and
+alias and_true_right := Bool.and_true
+alias not_and := Bool.and_deMorgan
+
+/-! ### or -/
+
+@[simp] theorem or_not_self_left : ∀ (x : Bool), (!x || x) = true := by
+  decide
+
+@[simp] theorem or_not_self_right : ∀ (x : Bool), (x || !x) = true := by
+  decide
+
+theorem or_comm : ∀ (x y : Bool), (x || y) = (y || x) := by
+  decide
+
+theorem or_left_comm : ∀ (x y z : Bool), (x || (y || z)) = (y || (x || z)) := by
+  decide
+
+theorem or_right_comm : ∀ (x y z : Bool), ((x || y) || z) = ((x || z) || y) := by
+  decide
+
+theorem or_and_distrib_left : ∀ (x y z : Bool), (x || (y && z)) = ((x || y) && (x || z)) := by
+  decide
+
+theorem or_and_distrib_right : ∀ (x y z : Bool), ((x && y) || z) = ((x || z) && (y || z)) := by
+  decide
+
+theorem or_deMorgan : ∀ (x y : Bool), (!(x || y)) = (!x && !y) := by
+  decide
+
+theorem or_eq_true_iff : ∀ (x y : Bool), (x || y) = true ↔ x = true ∨ y = true := by
+  decide
+
+theorem or_eq_false_iff : ∀ (x y : Bool), (x || y) = false ↔ x = false ∧ y = false := by
+  decide
+
+alias or_false_left := Bool.false_or
+alias or_false_right := Bool.or_false
+alias or_true_left := Bool.true_or
+alias or_true_right := Bool.or_true
+alias not_or := Bool.or_deMorgan
+
+/-! ### xor -/
+
+@[simp] theorem xor_false_left : ∀ (x : Bool), xor false x = x := by
+  decide
+
+@[simp] theorem xor_false_right : ∀ (x : Bool), xor x false = x := by
+  decide
+
+@[simp] theorem xor_true_left : ∀ (x : Bool), xor true x = !x := by
+  decide
+
+@[simp] theorem xor_true_right : ∀ (x : Bool), xor x true = !x := by
+  decide
+
+@[simp] theorem xor_not_self_left : ∀ (x : Bool), xor (!x) x = true := by
+  decide
+
+@[simp] theorem xor_not_self_right : ∀ (x : Bool), xor x (!x) = true := by
+  decide
+
+@[simp] theorem xor_not_left : ∀ (x y : Bool), xor (!x) y = !(xor x y) := by
+  decide
+
+@[simp] theorem xor_not_right : ∀ (x y : Bool), xor x (!y) = !(xor x y) := by
+  decide
+
+theorem xor_not_not : ∀ (x y : Bool), xor (!x) (!y) = (xor x y) := by
+  decide
+
+theorem xor_self : ∀ (x : Bool), xor x x = false := by
+  decide
+
+theorem xor_comm : ∀ (x y : Bool), xor x y = xor y x := by
+  decide
+
+theorem xor_left_comm : ∀ (x y z : Bool), xor x (xor y z) = xor y (xor x z) := by
+  decide
+
+theorem xor_right_comm : ∀ (x y z : Bool), xor (xor x y) z = xor (xor x z) y := by
+  decide
```

Comment:
Could you say why `xor_left_comm` and `xor_right_comm` are useful?  Is it for normalization purposes?

Reply:
These are "traditional" (cf. `Nat.add_right_comm`, `Nat.mul_right_comm`, etc.) They save some juggling parentheses here and there. Long ago, some of these were marked simp as a safer way to use commutativity in simp. This is not the case anymore but they are probably still used in some normalization procedures. @digama0 probably knows the history better than I do.

Reply:
Thanks!